.github/workflows/nomis.yml

```diff
@@ -1,58 +1,366 @@
+
+      - name: Setup project matrix
+        id: setupproject
         env:
-          TF_ENV: production
+          projects: ${{ steps.parseinput.outputs.projects }}
+        run: |
+          echo "Setup ansible actions for projects [${projects}]"
+          echo -n "::set-output name=matrix::{\"include\":["
+          delimiter=""
+          for project in ${projects}; do
+            echo -n "${delimiter}"
+            echo -n '{"project":"'${project}'"}'
+            delimiter=","
+          done
+          echo "]}"
+
+      # security best practice: use specific version of public actions (v0.3.0)
+      - name: Hide previous PR comments
+        uses: int128/hide-comment-action@24848bfab99f16e129632e27397da999b9d1b2c0
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          authors: ""
+          starts-with: "**`${{ env.TEAM_DIR }}/"
+
 
-  deploy-nomis:
-    name: Deploy - nomis pipelines
-    needs: plan-nomis
+  plan:
+    needs: init
+    if: ${{ needs.init.outputs.projects != '' }}
+    strategy:
+      matrix: ${{ fromJson(needs.init.outputs.matrix) }}
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+
+      - name: Get terraform version
+        id: discover
+        run: |
+          echo "Get terraform version for ${{ matrix.project }}"
+          set +o pipefail
+          required_version=$(grep ^terraform "${TEAM_DIR}/${{ matrix.project }}"/*.tf -A 20 | grep -w required_version | cut -d\" -f2)
+          if [[ -z "$required_version" ]]; then
+            echo "Using default terraform version specified in pipeline $DEFAULT_TERRAFORM_VERSION" >&2
+            required_version="=${DEFAULT_TERRAFORM_VERSION}"
+          fi
+          if [[ ! "$required_version" =~ =[0-9]+.[0-9]+.[0-9]+ ]]; then
+            echo "Unexpected terraform required_version format, expect '=x.y.z': $required_version" >&2
+            exit 1
+          fi
+          version=$(echo "$required_version" | cut -d= -f2)
+          echo "version=${version}"
+          echo ::set-output name=version::${version}
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ steps.discover.outputs.version }}
+          terraform_wrapper: false
+
+      # security best practice: use specific version of public actions (v1)
+      - name: Turnstyle
+        id: turnstyle
+        uses: softprops/turnstyle@8db075d65b19bf94e6e8687b504db69938dc3c65
+        with:
+          same-branch-only: false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Init and State Refresh
+        working-directory: "${{ env.TEAM_DIR }}/${{ matrix.project }}"
+        run: |
+          ${GITHUB_WORKSPACE}/scripts/terraform-init.sh .
+          terraform workspace select "core-shared-services-${TF_ENV}"
```

Comment:
`terraform workspace select "core-shared-services-${TF_ENV}" || terraform workspace new "core-shared-services-${TF_ENV}"`

The workspace won't exist when a new state is created so it'll save hacking about with the workflow file every time a new pipeline's added.

Reply:
Totally agree - but would like to put that in the next PR, where I will create a new terraform state, to ensure all of that is fully tested.  There maybe other tweaks required as well